internal/clients/http/ec2/ec2_client.go

```diff
@@ -26,31 +27,62 @@ type EC2Client struct {
 }
 
 func init() {
-	clients.GetEC2Client = NewEC2Client
-	clients.GetEC2ClientWithRegion = NewEC2ClientWithRegion
+	clients.GetAssumedEC2Client = newAssumedEC2Client
+	clients.GetAssumedEC2ClientWithRegion = newAssumedEC2ClientWithRegion
 }
 
-func NewEC2Client(ctx context.Context) (clients.EC2, error) {
-	c, _ := NewEC2ClientWithRegion(ctx, config.AWS.Region)
-	return c, nil
+func newAssumedEC2Client(ctx context.Context, arn string) (clients.EC2, error) {
+	return newAssumedEC2ClientWithRegion(ctx, arn, config.AWS.Region)
 }
 
-func NewEC2ClientWithRegion(ctx context.Context, region string) (clients.EC2, error) {
-	c := &EC2Client{
-		context: ctx,
-		log:     ctxval.Logger(ctx).With().Str("client", "ec2").Logger(),
+func newAssumedEC2ClientWithRegion(ctx context.Context, arn string, region string) (clients.EC2, error) {
+	logger := ctxval.Logger(ctx)
+
+	creds, err := getStsAssumedCredentials(ctx, arn)
+	if err != nil {
+		return nil, err
+	}
+
+	newCfg, err := awsCfg.LoadDefaultConfig(ctx, awsCfg.WithRegion(region), awsCfg.WithCredentialsProvider(
+		credentials.NewStaticCredentialsProvider(*creds.AccessKeyId, *creds.SecretAccessKey, *creds.SessionToken),
+	))
+	if err != nil {
+		return nil, fmt.Errorf("cannot create a new ec2 config: %w", err)
 	}
 
-	log.Trace().Msg("Creating new EC2 client")
-	cache := aws.NewCredentialsCache(credentials.NewStaticCredentialsProvider(
-		config.AWS.Key, config.AWS.Secret, config.AWS.Session))
+	return &EC2Client{
+		ec2:     ec2.NewFromConfig(newCfg),
+		context: ctx,
+		log:     logger.With().Str("client", "ec2").Logger(),
```

Comment:
We should probably leave a field in the logger, something like "client_type" with values "service" or "customer" to differentiate between the two.

Reply:
if we have both then yes, but I'm removing the Service one here, we can bring it back, but given both were kinda incorect, I'd rather do one right in this PR and add the Service one later. I can rename the Assumed to Customer tho, thats a good idea anyway :)